src/v/kafka/server/protocol_utils.cc

```diff
@@ -71,8 +71,8 @@ parse_header(ss::input_stream<char>& src) {
 }
 
 size_t parse_size_buffer(ss::temporary_buffer<char>& buf) {
-    auto* raw = ss::unaligned_cast<const int32_t*>(buf.get());
-    int32_t size = be_to_cpu(*raw);
+    auto* raw = reinterpret_cast<const int32_t*>(buf.get());
+    int32_t size = ss::be_to_cpu(*raw);
```

Comment:
I think this still falls foul of strict aliasing rules, I'm under the impression that the bytes must be copied. I.e., the dereference of `raw` is not allowed.

Reply:
Looking through the file, I think there are several places we're violating the spec.  Let's address that separately to the seastar compatibility update: https://github.com/vectorizedio/redpanda/issues/2916